### ISSUE TITLE
feat: add config path option

### ISSUE THREAD

**Issue by tom-fletcher** [JamieMason/syncpack#71]
## Description

Currently, `syncpack` only locates the config file (if present) with `cosmiconfig`.

Other tools, such as ESLint and Prettier, allow the user to specific a configuration path other than the normal one auto-detected. This can be useful for i.e. moving the config files out of the package root and into a `config/` folder. Allowing a custom config path would bring `syncpack` into parity with these tools.

## Suggested Solution

Add `-c` / `--config` option to CLI:

```diff
  -i, --indent [value]    override indentation. defaults to "  "
+ -c, --config <path>     path to a syncpack config file, overriding automatic detection
  -h, --help              display help for command
```

When specified, use that path to locate the config file. If the file is not present, exit with an error.

When not specified, locate the config settings with `cosmiconfig` as normal.

## PR

Happy to look at submitting a PR on this.

**Comment by tom-fletcher** [comment]
I ended up with a bit of time this afternoon to look at this, and it was fairly straightforward. `cosmiconfig` does the heavy lifting again, all that was needed was to add a `config` option to the CLI and pass this through.

Of course still open for discussion if there are opinions on how to implement!

I decided to pass the config path through as a new parameter on `getInput()`

```diff
export function getInput(
   disk: Disk,
+  configPath: string | undefined, 
   program: Partial<SyncpackConfig>,
): ProgramInput
```

Initially I considered passing this through with other options on the `program` parameter, but this would have meant adding a `config:` key to the `SyncpackConfig` type, and it felt better to seperate this. The `config` option points **to** the config, rather than being **part of** the config. This did mean some tests needed to be updated, just to pass an extra option through.
